### PR TITLE
Navbar fixes

### PR DESCRIPTION
**Pull request by yrammos**
The list of users in the two navigation bars (Admin and Profile) is not always synchronized with the actual list of users if two or more users are active. This PR fixes the issue.

The fix consists of two parts. First, only the currently active user is removed from `profileList[]` (I am avoiding `slice()` for this purpose because it is considered bad ReactJS practice on array-like states). Second, `this.fetchAPI()` is no longer called if a user has just logged out.

An edge case producing a crash is also fixed by ensuring that `this.state.profileList[0]` exist before attempting to index it.

To test, please log in with at least two usernames, then log them out one by one, each time noticing the effects on the navigation bar dropdowns.

### CODE REVIEW

**Comment by yrammos** [comment]
Hmm… the revision solves one problem by creating another (it breaks the green "Continue" button in certain scenarios).

Will revise and reopen. Closing for now.

**Comment by babelouest** [comment]
I think this can be solved in another way, please see the code comments

File: webapp-src/src/Admin/App.js [hunk 1]
```diff
@@ -104,6 +104,8 @@ class App extends Component {
         this.setState({loggedIn: message.message}, () => {
```
Comment (babelouest) [inline webapp-src/src/Admin/App.js:104]: you can try by changing the whole block like this:

```javascript
this.setState({loggedIn: message.loggedIn}, () => {
  this.fetchApi();
});
```

File: webapp-src/src/Profile/App.js [hunk 2]
```diff
@@ -64,7 +64,8 @@ class App extends Component {
       } else if (message.type === 'loggedIn') {
         this.setState({loggedIn: message.message}, () => {
```
Comment (babelouest) [inline webapp-src/src/Profile/App.js:65]: Like above, this block can be replaced by this:

```javascript
this.setState({loggedIn: message.loggedIn}, () => {
  this.fetchProfile();
});
```

**Comment by yrammos** [comment]
@babelouest I found an other, more verbose solution meanwhile, but will now try your more compact suggestion and report back.

File: webapp-src/src/Admin/App.js [hunk 3]
```diff
@@ -102,10 +102,10 @@ class App extends Component {
         this.fetchApi();
       } else if (message.type === 'loggedIn') {
         this.setState({loggedIn: message.message}, () => {
-          console.log("loggedIn", this.state.loggedIn);
           if (!this.state.loggedIn) {
-            this.fetchApi();
+            this.setState({curNav: "users",profileList: this.state.profileList.filter((e,i) => i !== 0)});
```
Comment (babelouest) [inline webapp-src/src/Admin/App.js:106]: This isn't required anymore, the `this.fetchApi();` below will refresh the user list from the backend, and dispatch the new list to all the subcomponents

File: webapp-src/src/Profile/App.js [hunk 4]
```diff
@@ -64,10 +64,10 @@ class App extends Component {
...
           if (!this.state.loggedIn) {
-            this.setState({profileList: false, schemeList: [], curNav: "profile"});
-          } else {
-            this.fetchProfile();
+            // Not sure about `schemeList` in the line below!
+            this.setState({curNav: "profile", profileList: this.state.profileList.filter((e,i) => i !== 0), schemeList: []});
```
Comment (babelouest) [inline webapp-src/src/Profile/App.js:68]: This isn't required anymore, the `this.fetchProfile();` below will refresh the user list from the backend, and dispatch the new list to all the subcomponents

**Comment by babelouest** [comment]
@yrammos , when you push changes in a PR, please don't push force a rebased branch, simply add new commits on top of the last one, otherwise it's harder to follow up changes.

Also, when a required change is commited, you should use the `Resolve conversation` button each time, likewise we'll know where we're at without having to re-read all the code changes.

**Comment by yrammos** [comment]
@babelouest, sorry about the rebase (symptom of a mostly solo worker).

I implemented your change but the issue with the *Admin* navbar persisted: the last user to log out would still appear on the navbar. I tracked it down to a tiny glitch with `this.fetchApi()`. Kindly review for any edge cases I may have not considered.

**Comment by babelouest** [comment]
@yrammos , I've tested the PR and I've seen no edge case or new bugs. I'm pushing as is, thanks!
